### ISSUE TITLE
space-before-keywords misrecognizes shorthand methods as keywords

### ISSUE THREAD

**Issue by mzgol** [eslint/eslint#3756]
The `space-before-keywords` rule (in ESLint 1.4.1) misrecognizes shorthand methods as keywords:
```js
let x = {a() {}};
```
It claims `a` is a keyword and should be preceded by a space.

I want to not write spaces inside the parens in one-line blocks or arrays:
```js
const a = [1, 2, 3];
const o = {a: 2, b: 3};
```
but I still want spaces after & before keywords:
```js
if (x) {
  /* ... */
} else {
  /* ... */
}
```

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by SpaceK33z** [comment]
It also returns a warning when using a class that extends an object:

```js
class Foo extends Marionette.Object { }
```

Returns: `Missing space before keyword "."`

This is a somewhat related to your issue, so I added it here instead of creating a new issue.

**Comment by mmrko** [comment]
Uh oh, thanks for bringing these up! Looks like I've missed a couple special cases. I'll investigate this further.
